1] guessing game/guessing_game.py

```diff
@@ -0,0 +1,64 @@
+"""
+Workshop_01 - Program_01
+--------------------
+Number Guessing Game
+--------------------
+(c) Tomas Dolejsek 2024-01-30
+
+Write a simple number guessing game. The computer must draw a number in the range 1 – 100. Then it should:
+
+1. Ask: "Guess the number: " and retrieve the number from the keyboard.
+2. Check whether it is really a number and in case of an error display the message "It's not a number!",
+and then return to point 1.
+3. If the number entered by the user is smaller than the drawn number, display the message "Too small!",
+and return to point 1.
+4. If the number entered by the user is greater than the drawn number, display the message "Too big!",
+and then return to point 1.
+5. If the number entered by the user is equal to the drawn number, display the message "You win!",
+and then terminate the program.
+"""
+
+from random import randint
+
+
+class OutOfRangeError(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+
+
+def get_number() -> int:
+    """
+    Take an input from the user and validate its type (int) and range (0-100).
+    :return: validated number
+    """
+    while True:
+        try:
+            num = int(input("Guess a number (1 - 100): "))
+            if not (1 <= num <= 100):
+                raise OutOfRangeError("The number is out of range!")
```

Comment:
jesli chces delat classu na OutOfRange error tak uz by uvnitr te klasy mela byt logika ktera ten range ocekuje, at to nemusi byt tady